.github/actions/dependencies/action.yml

```diff
@@ -17,7 +17,7 @@ runs:
     - name: install dependencies
       shell: bash
       run: |
-        mkdir ${build_dir}
+        mkdir -p ${build_dir}
```

Comment:
Was this actually necessary? I don't think the build directory has multiple steps, so I'm guessing this was added to get around an existing build directory. I suspect we should instead remove and remake the build directory if that was the reason.

Reply:
I _think_ this was to work around an existing directory. IIRC, the workflow has a "build dependencies" step, which calls this script directly *and* a "build" step, which calls this workflow indirectly. It must have been making the second step fail.

I'll go ahead and remove the change, and see how that affects things.

Reply:
@ximinez are you working with your local windows machine or tinkering with the Windows runner provided by Github Actions?

I don't have a windows machine and I'm wondering how I can try out these build steps.

Reply:
Yep, that was it. https://github.com/ximinez/rippled/actions/runs/6343277082/job/17230820510#step:13:61

The reason it's not an issue on the `nix` jobs is that those two steps are split across the separate jobs. My problem is a side effect of bringing them back together. I can think of three ways to make this work.
1. Put the `-p` back with a comment explaining that it's for this scenario.
2. `rm -rf ${build_dir}` before creating it to ensure we're starting with a fresh slate.
3. `[ -d ${build_dir} ] || mkdir ${build_dir}` so it doesn't error out if it exists.

I don't really have a strong feeling one way or another. 2 is probably the safest option, even though it'll add a little time to the build because it'll need to recreate the conan toolchain file. What do you think?

Reply:
> @ximinez are you working with your local windows machine or tinkering with the Windows runner provided by Github Actions?

This is with the Windows runner from Github Actions.

> I don't have a windows machine and I'm wondering how I can try out these build steps.

You could copy my branch into a branch in your own repo and play with it from there.

Reply:
okay, I'll check it out 👍

Reply:
I think the right fix is to remove the link between the `build` and `dependencies` actions, and just add a step for the `dependencies` action to the `test` job in the `nix` workflow.

Alternatively, but not my preference, you could remove the `dependencies` step in the `windows` workflow and move the upload step after the `build` step, adding an `always() &&` clause to its `if` condition.